File: Sorting/Selection Sort/Selectionsort.py

# Selection sort in Python
def selectionSort(array, size):
    for i in range(size):
        minpos = i
        for j in range(i+1, size):
            if array[j] < array[minpos]:
                minpos = j
        (array[i], array[minpos]) = (array[minpos], array[i])
arr=[]
n=int(input())
for i in range(0,n):
 element=int(input())
 arr.append(element)
size = len(arr)
selectionSort(arr, size)
print('array after sort:')
print(arr)

File: Sorting/Selection Sort/Selection sort.cpp
//Selection sort in C++
#include<iostream>
using namespace std;
int main()
{
    int n;
    cin>>n;
    int a[n];
    for(int i=0;i<n;i++)
    cin>>a[i];
    for(int i=0;i<n;i++)
    {
        int minpos=i;
        for(int j=i;j<n;j++)
        {
            if(a[j]<a[minpos])
            minpos=j;
        }
        swap(a[i],a[minpos]);
    }
    for(int i=0;i<n;i++)
    cout<<a[i]<<" ";
    return 0;
}
